dapp-core/src/test/kotlin/jp/co/soramitsu/dapp/test/environment/DappTestEnvironment.kt

```diff
@@ -0,0 +1,220 @@
+    FixedHostPortGenericContainer<KGenericFixedContainer>(imageName)
+
+private const val DEFAULT_RMQ_PORT = 5672
+private const val DEFAULT_IROHA_PORT = 50051
+const val dappDomain = "dapp"
+const val dappInstanceAccountId = "dapp1" + Const.accountIdDelimiter + dappDomain
+const val dappRepoAccountId = "dapprepo" + Const.accountIdDelimiter + dappDomain
+val irohaKeyPair = Ed25519Sha3().generateKeypair()!!
+val rmq = KGenericFixedContainer("rabbitmq:3-management").withExposedPorts(DEFAULT_RMQ_PORT)
+    .withFixedExposedPort(DEFAULT_RMQ_PORT, DEFAULT_RMQ_PORT)
+    .withCreateContainerCmdModifier { it.withName("rmq") }!!
+val iroha = IrohaContainer().withPeerConfig(peerConfig)!!
+var postgresDockerContainer: GenericContainer<*> = GenericContainer<Nothing>()
+var irohaContainer: GenericContainer<*> = GenericContainer<Nothing>()
+val chainAdapter = KGenericFixedContainer("nexus.iroha.tech:19002/d3-deploy/chain-adapter:1.0.0_rc5")
+const val rmqExchange = "iroha"
+const val resourcesLocation = "src/test/resources"
+
+val logger = KLogging().logger
+
+val genesisBlock: BlockOuterClass.Block
+    get() {
+        try {
+            return GenesisBlockBuilder()
+                .addDefaultTransaction()
+                .addTransaction(
+                    Transaction.builder(dappRepoAccountId)
+                        .createDomain(dappDomain, GenesisBlockBuilder.defaultRoleName)
+                        .createAccount(
+                            dappInstanceAccountId,
+                            irohaKeyPair.public
+                        )
+                        .createAccount(
+                            dappRepoAccountId,
+                            irohaKeyPair.public
+                        )
+                        .setAccountDetail(
+                            dappRepoAccountId, "testcontract", Utils.irohaEscape(
+                                Files
+                                    .toString(
+                                        File("src/test/resources/sample_contract.groovy"),
+                                        Charset.defaultCharset()
+                                    )
+                            )
+                        )
+                        .setAccountDetail(dappInstanceAccountId, "testcontract", "true")
+                        .createAsset("asset", dappDomain, 2)
+                        .build()
+                        .build()
+                )
+                .build()
+        } catch (e: IOException) {
+            logger.error("Genesis block building exception occurred", e)
+            throw RuntimeException(e)
+        }
+    }
+
+val peerConfig: PeerConfig
+    get() = PeerConfig.builder()
+        .genesisBlock(genesisBlock)
+        .build()
+
+class DappTestEnvironment : Closeable {
+
+    lateinit var service: DappService
+    lateinit var irohaAPI: IrohaAPI
+    lateinit var queryAPI: QueryAPI
+    private lateinit var rmqHost: String
+    private var rmqPort: Int = 0
```

Comment:
Why 0?

Reply:
since `lateinit` is not allowed for primitives